Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -177,6 +180,34 @@ void NetKDRequestDevice::Update()
   }
 }
 
+void NetKDRequestDevice::LogError(ErrorType error_type, s32 error_code)
+{
+  s32 new_code{};
+  switch (error_type)
+  {
+  case ErrorType::ACCOUNT:
+    new_code = -(101200 - error_code);
+    break;
+  case ErrorType::CLIENT:
+    new_code = -(107300 - error_code);
+    break;
+  case ErrorType::KD_DOWNLOAD:
+    new_code = -(107200 - error_code);
+    break;
+  case ErrorType::SERVER:
+    new_code = -(117000 + error_code);
+    break;
+  }
+
+  std::lock_guard lg(m_scheduler_buffer_lock);
+
+  m_scheduler_buffer[32 + m_error_count] = Common::swap32(new_code);
+  m_error_count++;
```

Comment:
This never seems to be reset, so this might go out of bounds over time I think?

Reply:
That is true, best practice should be to check if it will go out of bounds. However, no official title will call KD Download enough, let alone resume after an error to cause a bounds error.

Reply:
Looking at actual behaviour, it will reset the index back to 0 once there are 32 errors, but keep the error count.